crt_portal/static/js/redirect-modal.js

```diff
@@ -0,0 +1,33 @@
+(function(root) {
+  // note that modal.js must be loaded beforehand
+  var modal_el = document.getElementById('external-link--modal');
+  var span = document.getElementById('external-link--address');
+  var links = document.querySelectorAll('.external-link--popup');
+  var continue_button = document.getElementById('external-link--continue');
+  var redirect;
+  for (var i = 0; i < links.length; i++) {
+    var link = links[i];
+    link.onclick = function(event) {
+      event.preventDefault();
+      // display the actual redirect link
+      span.innerHTML = '<a href="' + link.href + '">' + link.href + '</a>';
+      root.CRT.openModal(modal_el);
+      // set timeout for redirect
+      clearTimeout(redirect);
+      redirect = setTimeout(function() {
+        // only redirect if modal is still visible
+        if (modal_el.getAttribute('hidden') === null) {
+          window.location.href = link.href;
+        }
+      }, 20000);
+
+      // set up "continue" button to immediately redirect
+      continue_button.onclick = function(event) {
+        event.preventDefault();
+        window.location.href = link.href;
+      };
```

Comment:
I am seeing an issue when clicking lsc.gov/find-legal-aid link I shows me the http://www.findlegalhelp.org/ dialog instead.

![image](https://user-images.githubusercontent.com/66343959/89658982-36dade80-d89d-11ea-9b5c-c86a0a316243.png)

This issue seems to be caused by not setting the link value in your click function. Because of this your link value is always going to return your last link element set on line 9. I recommend setting the link value inside the click function. You can do something like this `link = event.target;` and it should fix the issue. Other than that It looks great I was able to test everything else in the card.

Reply:
Gah, thank you :) I missed this one, fixing.

Reply:
Fixed -- thanks for the heads up. Closures in a loop don't work well in Javascript. D'oh.